### ISSUE TITLE
hide & deprecate enable-streaming

### ISSUE THREAD

**Issue by SunRunAway** [pingcap/tidb#20759]
## Feature Request

**Is your feature request related to a problem? Please describe:**

`enable-streaming` is not well maintained because we are focusing on optimizing the non-streaming coprocessor API for its performance and resource usage and don’t have time to optimize streaming's currently, I propose to deprecate it temporarily until somebody has time to pick it up again and prove it could be GA in the future.

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

see https://asktug.com/t/topic/37526/7

**Comment by zz-jason** [comment]
@SunRunAway could you link all the bugs that related to streaming API?

**Comment by SunRunAway** [comment]
See https://asktug.com/t/topic/37526 and https://asktug.com/t/topic/62984

**Comment by SunRunAway** [comment]
We are currently focusing on optimizing the non-streaming coprocessor API for its performance and resource usage, but when the users enable the steaming API, they will not be benefited from us.

**Comment by breeswish** [comment]
@zz-jason  There are a lot of known trivial issues, for example, 

1. TiKV known bug: https://github.com/tikv/tikv/issues/3677 (won't fix)
2. TiKV does not handle streaming scan efficiently
3. Coprocessor Cache does not work in streaming
4. The runtime statistics in streaming mode is not correct. The newly added ScanDetailV2 may be incorrect as well.
5. When I reviewed the Coprocessor part in TiDB I found that it may not correctly handle the case that partial result is lost and request is retried. There is no test case for this either.
